token/cli/src/config.rs

```diff
@@ -276,42 +322,52 @@ impl<'a> Config<'a> {
         override_name: &str,
         wallet_manager: &mut Option<Arc<RemoteWalletManager>>,
         token: Option<Pubkey>,
-    ) -> Pubkey {
-        if let Some(address) = pubkey_of_signer(arg_matches, override_name, wallet_manager).unwrap()
+    ) -> Result<Pubkey, Error> {
+        if let Some(address) = pubkey_of_signer(arg_matches, override_name, wallet_manager).ok().flatten()
         {
-            return address;
+            return Ok(address);
         }
 
         let token = token.unwrap();
         let program_id = self.get_mint_info(&token, None).await.unwrap().program_id;
-        self.associated_token_address_for_token_and_program(&token, &program_id)
+        let owner = self.maybe_pubkey(arg_matches, "owner", wallet_manager);
+        self.associated_token_address_for_token_and_program(&token, owner.as_ref(), &program_id).await
     }
 
-    pub(crate) fn associated_token_address_for_token_and_program(
+    pub(crate) async fn associated_token_address_for_token_and_program(
         &self,
         token: &Pubkey,
+        owner: Option<&Pubkey>,
```

Comment:
> Since this is only ever called in one other place, could you make this param non-optional, then fetch the `default_signer_pubkey()` and pass that in as the owner from the other call site?

Sure!

Reply:
Actually it's also called from main.rs ...

Reply:
Yes but the place where it's called in main.rs has no "owner" Pubkey to pass in.

```
                config.associated_token_address_for_token_and_program(
                    &mint_info.address,
                    None,
                    &mint_info.program_id,
                ).await?
```

Reply:
Right, so you'll have to fetch the owner with `let owner = config.default_signer.pubkey();` and pass that in. Does that make sense?

Reply:
Sure I can do that.  I'm a little confused why there is an "--owner" command line argument if the default signer can be treated as the owner, but I'll be happy to make the change.

Reply:
The concept is that if no owner is passed in, we use the default signer.  The chain of "Sensible Defaults" :tm: in the CLI can be really confusing

Reply:
> The concept is that if no owner is passed in, we use the default signer. The chain of "Sensible Defaults" tm in the CLI can be really confusing

OK I hope this latest change is sufficient.  The logic for acquiring the default_signer is a bit odd but without modifying the solana SDK it's the best I could do.